straxen/plugins/nveto_recorder.py

```diff
@@ -0,0 +1,374 @@
+import numba
+import numpy as np
+from scipy.ndimage import convolve1d
+from frozendict import frozendict
+
+import strax
+import straxen
+
+export, __all__ = strax.exporter()
+
+__all__ = ['nVETORecorder']
+
+
+@strax.takes_config(
+    strax.Option('coincidence_level', type=int, default=4,
+                 help="Required coincidence level."),
+    strax.Option('resolving_time', type=int, default=600,
+                 help="Resolving time of the coincidence in ns."),
+    strax.Option('baseline_samples_lone_records', type=int, default=10, track=False,
+                 help="Number of samples used in baseline rms calculation"),
+    strax.Option('n_lone_records', type=int, default=2, track=False,
+                 help="Number of lone hits to be stored per channel for diagnostic reasons."),
+    strax.Option('channel_map', track=False, type=frozendict,
+                 help="frozendict mapping subdetector to (min, max) "
+                      "channel number."),
+    strax.Option('n_nveto_pmts', type=int, track=False,
+        help='Number of nVETO PMTs'))
+class nVETORecorder(strax.Plugin):
+    __version__ = '0.0.3'
+    parallel = 'process'
+
+    rechunk_on_save = True
+    compressor = 'lz4'
+
+    depends_on = 'raw_records_prenv'
```

Comment:
About the nveto data_kinds pre_raw_records, raw_records and records. I think I should explain my thought behind the additional pre_raw_records level. I introduced this level since compared to the TPC we cannot store all of our data. In fact we will delete like 95 % of our recorded pulses. So I thought it would be best to have only for these remaining 5 % a data structure similar to the TPC in which I have for each record a raw_record etc.